### ISSUE TITLE
Support multiple readers

### ISSUE THREAD

**Issue by chkr1011** [aquaticus/esphome-iec62056#4]
I would like to connect multiple IR receivers to an ESP32 and propagate the values to Home Assistant.

Right now, I can define multiple UART components, but I cannot define multiple readers (see screenshot):

![image](https://github.com/user-attachments/assets/f9f36a7b-efa2-4ce9-9b46-f015439fa789)

**Comment by aquaticus** [comment]
Hi

The component does not support multiple instances.
This configuration won't work. You cannot connect multiple optical interfaces.

**Comment by chkr1011** [comment]
Is there a way to get this feature implemented in this component?

**Comment by aquaticus** [comment]
Yeah, it is. I started implementing it once, but I thought no one would ever use this feature. 😀

**Comment by chkr1011** [comment]
> Yeah, it is. I started implementing it once, but I thought no one would ever use this feature. 😀

Wow that would be great if this gets released. I agree that my situation is very special with 2 electrical meters and one water meter (in the future).

**Comment by chkr1011** [comment]
Do you have an ETA when this feature will be released?

**Comment by aquaticus** [comment]
No plans to implement this at the moment.

**Comment by samotion2** [comment]
I recently bought a few of your sensors, and I would also be interested in this.

**Comment by bangert** [comment]
This commit look interesting https://github.com/eringerli/esphome-iec62056/commit/db4b1086ca7c98fd7d012a66686591cc0f8adacb and related to this issue (without having tested this myself)

**Comment by aquaticus** [comment]
This option still does not exempt you from the obligation to modify the code in C. 😺

**Comment by aquaticus** [comment]
Multiple readers added in PR #7
